pythonforandroid/graph.py

```diff
@@ -99,17 +129,125 @@ def find_order(graph):
                 bset.discard(result)
 
 
-def get_recipe_order_and_bootstrap(ctx, names, bs=None):
-    recipes_to_load = set(names)
+def obvious_conflict_checker(ctx, name_tuples, blacklist=[]):
+    """ This is a pre-flight check function that will completely ignore
+        recipe order or choosing an actual value in any of the multiple
+        choice tuples/dependencies, and just do a very basic obvious
+        conflict check.
+    """
+    deps_were_added_by = dict()
+    deps = set()
+
+    # Add dependencies for all recipes:
+    to_be_added = [(name_tuple, None) for name_tuple in name_tuples]
+    while len(to_be_added) > 0:
+        current_to_be_added = list(to_be_added)
+        to_be_added = []
+        for (added_tuple, adding_recipe) in current_to_be_added:
+            if len(added_tuple) > 1:
+                # No obvious commitment in what to add, ignore it.
+                # (Remember this function only catches obvious issues)
+                continue
+
+            name = added_tuple[0]
+            recipe_conflicts = []
+            recipe_dependencies = []
+            try:
+                # Get recipe to add and who's ultimately adding it:
+                recipe = Recipe.get_recipe(name, ctx)
+                recipe_conflicts = [c.lower() for c in recipe.conflicts]
+                recipe_dependencies = \
```

Comment:
Python is not bash. :)

Reply:
@KeyWeeUsr this line isn't `>80` but pretty close without the backslash, and I considered it quite less readable that way. That is why this is one of the few places where I used it to increase readability. As usual, I'm happy to take it out if y'all want me to

Reply:
I think he meant to use parentheses over backslash

Reply:
you mean additional (syntactically non-significant) parentheses? hm, I guess that might be an idea, let me give it a try

Reply:
I don't know, that looks quite ugly and confusing to me:
```
                recipe_dependencies = (
                    get_dependency_tuple_list_for_recipe(
                        recipe, blacklist=blacklist
                    ))
```
Again this is one of the few instances where I think a backslash would actually make sense. And `pep8`/`tox` don't complain, so while I get it's a general rule to avoid it if possible, it's still meant to be used *when it makes sense*, right?

@KeyWeeUsr can you propose what you'd actually write?

Reply:
@JonasT The brackets approach doesn't mean you need to add more brackets :) Use already present code. Python is built on blocks with indentation, so if you see a bracket, it means you can do a block. From stuff such as
```python
int(
    1
)
```

up to your lines:
```python
                recipe_dependencies = get_dependency_tuple_list_for_recipe(
                    recipe, blacklist=blacklist
                )
```
< 79, no backslash, hopefully fully pep8 compliant. Backslash makes absolutely no sense here and it's ugly imo. I can't find an example where the backslash is pretty right now, however I can find e.g. >79 char line which is more readable than broken into multiple lines - https://github.com/kivy/pyjnius/blob/5f6aa738b2b7ae496ec43bfadbf9933ee640bb7b/jnius/jni.pxi#L117 because it's almost clean C and a function pointer + a table-like structuring of a file to easily find return type, etc.

Reply:
@KeyWeeUsr I find that to be quite less readable, however. Because it's indented so much and `get_dependency_tuple_list_for_recipe` is such a long name, it ends up looking quite cramped. Again if you really want me to change it I'll do it, but I *personally* think it's uglier / less readable here.

So what's your final call?

Reply:
@KeyWeeUsr changed it now as per your final call in chat :+1: this is really not a hill I want to die on so that's fine with me, it was really just what I legitimately thought looked better but that is obviously very subjective :grimacing:

Reply:
I hope you're fine with me resolving this, since I added in your suggestion :+1: and it was an insightful discussion for sure!